### PR TITLE
Decorate classes

### PR DESCRIPTION
**Pull request by AbbyNyakara**
In this PR, I used a class decorator for my person class to add extra functioality to it.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,26 +1,67 @@
...
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < NameableBaseDecorator
+  def correct_name
+    @nameable.correct_name
+    @nameable.correct_name[0...10] if @nameable.correct_name.length > 10
+  end
+end
```
Comment (TedLivist) [inline person.rb]: - [x] This method returns `nil` for names less than 10 in length. Kindly fix that.

  **SUGGESTION:**
  ```
  def correct_name
    name = @nameable.correct_name.length > 10 ? @nameable.correct_name[0...10] : @nameable.correct_name
    name
  end
  ```
  ma

- [x] **[OPTIONAL]** I would recommend you move this class to its own file in keeping with Ruby conventions.

File: person.rb [hunk 2]
```diff
@@ -1,26 +1,67 @@
...
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < NameableBaseDecorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
```
Comment (TedLivist) [inline person.rb]: - [ ] **[OPTIONAL]** I would recommend you move this class to it own file in keeping with Ruby conventions

File: person.rb [hunk 3]
```diff
@@ -1,26 +1,67 @@
...
+
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```
Comment (TedLivist) [inline person.rb]: - [x] **[OPTIONAL]** I would recommend you move this class to its own file in keeping with Ruby conventions

File: person.rb [hunk 4]
```diff
@@ -1,26 +1,67 @@
-class Person
+class Nameable
+  def correct_name
+    raise NotImplementedError
+  end
+end
```
Comment (TedLivist) [inline person.rb]: - [x] **[OPTIONAL]** I would recommend you move this class to its own file in keeping with Ruby conventions

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @AbbyNyakara,

Good job so far!
✔️ Implementation of classes
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [x] **Kindly format your README using [the Microverse template](https://github.com/microverseinc/readme-template)**

- [x] _Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 4 above)
Comment (AbbyNyakara) [reply person.rb]: Done

**Review by omarsalem7 — APPROVED** [review]
Hi @AbbyNyakara 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**